nestkernel/scheduler.cpp

```diff
@@ -567,6 +569,23 @@ nest::Scheduler::update()
       if ( print_time_ )
         gettimeofday( &t_slice_begin_, NULL );
 
+      if ( net_->structural_plasticity_enabled_
+        && ( clock_.get_steps() + from_step_ ) % SPManager::structural_plasticity_update_interval
+          == 0 )
+      {
+        for ( i = nodes_vec_[ t ].begin(); i != nodes_vec_[ t ].end(); ++i )
+          ( *i )->update_synaptic_elements(
+            Time( Time::step( clock_.get_steps() + from_step_ ) ).get_ms() );
+#pragma omp barrier
+#pragma omp single
+        {
```

Comment:
Why do we need the `omp barrier` and `single`? The method `Scheduler::init_()` is private and only called in the constructor and in `reset()`, which are also not called in an `omp parallel` region.